### ISSUE TITLE
Function rng_get_max_int makes non-portable assumptions

### ISSUE THREAD

**Issue by bska** [Ensembles/ert#848]
Commit 39003a1 (PR #847) introduced a new function,
```C
unsigned int rng_get_max_int(const rng_type * rng)
```
The implementation of this function contains the statement
```C
unsigned int MAX = ((1 << 32) - 1);
```
which generates undefined behaviour.

In particular, this statement assumes that

1. The type `unsigned int` is at least 32 bits (common)
2. The type `int` is at least **33** bits (very rare, even on 64 bit architectures)

Even if one tries to work around that last assumption by writing
```C
1U << 32
```
some compilers still generate warnings of the form
```
/home/bska/work/opm/src/3rdparty/ert/devel/libert_util/src/rng.c:234:5: warning: left shift count >= width of type [enabled by default]
     unsigned int MAX = (1U << 32) - 1;
     ^
```

This problem can be fixed in a portable way by writing
```C
unsigned int MAX = -1;
```
instead.

**Comment by joakim-hove** [comment]
> The type int is at least 33 bits (very rare, even on 64 bit architectures)

Come on - don't be so conservative; let's make some bold and fresh assumptions! 

Joke aside  - thank you for the detailed bug report and the suggested fix. This has been fixed as the first commit of: https://github.com/Ensembles/ert/pull/849

Side note: Observe the brand new integration with http://travis-ci.org; quite nifty functionality which we could implement in OPM as well.

**Comment by bska** [comment]
> Observe the brand new integration with http://travis-ci.org; quite nifty functionality which we could implement in OPM as well.

Thank you for the pointer.  I'll have a look later.

**Comment by bska** [comment]
> This has been fixed as the first commit of: #849

Confirmed.  I'll close this.
